app/models/favorite.rb

```diff
@@ -0,0 +1,18 @@
+class Favorite
+  attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || Array.new
+  end
+
+  def total_count
+    @contents.length
+  end
+
+  def add_pet(id)
+    if !@contents.include?(id)
+      @contents << id
+    end
+  end
+
+end
```

Comment:
nice work driving. i'm glad we chose an array and it went in nicely. i think we'll be fine using a hash later when we need to.